native/game_backend/src/player.rs

```diff
@@ -342,6 +359,20 @@ impl Player {
             .iter()
             .any(|action_tracker| action_tracker.action != Action::Moving)
     }
+
+    pub fn can_perform_attack(&self, skill_key: &String) -> bool {
+        // Check if player is still performing an action or if skill is still on cooldown.
+        self.can_do_action()
+            && (can_perform_basic_attack(self) || can_perform_skill_attack(self, skill_key))
+    }
+}
+
+fn can_perform_basic_attack(player: &Player) -> bool {
+    player.cooldowns.contains_key(BASIC_SKILL_KEY) && player.available_burst_loads > 0
+}
+
+fn can_perform_skill_attack(player: &Player, skill_key: &String) -> bool {
+    !player.cooldowns.contains_key(BASIC_SKILL_KEY) && !player.cooldowns.contains_key(skill_key)
```

Comment:
Why are we checking that the player doesn't have a cooldown for the basic_skill_key here?

Reply:
Actually that's unnecessary, I changed it to a check against the BASIC_SKILL_KEY constant

Reply:
Nice!